software/py/nbf.py

```diff
@@ -243,42 +242,45 @@ def init_vcache(self):
  
   # init DRAM
   def init_dram(self, enable_dram): 
-    y = self.num_tiles_y
-    dram_ch_size = self.dram_ch_size
+    #y = self.num_tiles_y
     cache_size = self.cache_size
     lg_x = self.safe_clog2(self.num_tiles_x)
     lg_block_size = self.safe_clog2(self.cache_block_size)
     lg_set = self.safe_clog2(self.cache_set)
-    index_width = self.addr_width-1-lg_block_size
+    index_width = self.addr_width-1-lg_block_size-1
 
     if enable_dram == 1:
-      if self.num_tiles_x == 9:      
-        # hashing for 9 banks
-        for k in sorted(self.dram_data.keys()):
-          addr = k - 0x20000000
-          bit_2_0 = (addr >> lg_block_size) & 0b111
-          bit_5_4 = (addr >> (3+lg_block_size)) & 0b110
-          bit_3 = (addr >> (3+lg_block_size)) & 0b001
-          bit_12 = (addr >> (3+lg_block_size+lg_set)) & 0b1
-
-          if bit_2_0 == (bit_5_4 | (bit_3 ^ bit_12)):
-            x = 8
-          else:
-            x = bit_2_0
+      #if self.num_tiles_x == 9:      
+        # hashing for 9 banks (deprecated)
+      #  for k in sorted(self.dram_data.keys()):
+      #    addr = k - 0x20000000
+      #    bit_2_0 = (addr >> lg_block_size) & 0b111
+      #    bit_5_4 = (addr >> (3+lg_block_size)) & 0b110
+      #    bit_3 = (addr >> (3+lg_block_size)) & 0b001
+      #    bit_12 = (addr >> (3+lg_block_size+lg_set)) & 0b1
+
+      #    if bit_2_0 == (bit_5_4 | (bit_3 ^ bit_12)):
+      #      x = 8
+      #    else:
+      #      x = bit_2_0
     
-          block_offset = self.select_bits(addr, 0, lg_block_size-1)
-          index = self.select_bits(addr, lg_block_size+3, lg_block_size+3+index_width-1) << lg_block_size
-          epa = block_offset | index
-          self.print_nbf(x, y, epa, self.dram_data[k])
+      #    block_offset = self.select_bits(addr, 0, lg_block_size-1)
+      #    index = self.select_bits(addr, lg_block_size+3, lg_block_size+3+index_width-1) << lg_block_size
+      #    epa = block_offset | index
+      #    self.print_nbf(x, y, epa, self.dram_data[k])
```

Comment:
Do we anticipate using the commented deprecated code later?

Reply:
if we decide to make another 9x15 chip.

Reply:
Then why comment it out?

Reply:
Are you suggesting that we should just remove this instead of commenting out?

Reply:
initially, yes. I've since been convinced and I now understand the utility of keeping it. However, if we're keeping it, why comment it out? 

If it's un-commented, if someone ever modifies the code it's their responsibility to keep it up to date. If it's commented, it's whoever un-deprecates it. 

Plus, it's a large block of code. Leaving in a commented block of code is unsightly.

For example, if we don't want it in the code, it could be put in a separate function and then that function can be commented.

Reply:
I will just move it to the end of the class.